packages/firestore/src/local/local_store.ts

```diff
@@ -816,32 +815,31 @@ export class LocalStore {
   releaseQuery(query: Query, keepPersistedQueryData: boolean): Promise<void> {
     const mode = keepPersistedQueryData ? 'readwrite' : 'readwrite-primary';
     return this.persistence.runTransaction('Release query', mode, txn => {
-      return this.getQueryData(txn, query).next(queryData => {
-        assert(
-          queryData != null,
-          'Tried to release nonexistent query: ' + query
-        );
-        const targetId = queryData!.targetId;
-
-        // References for documents sent via Watch are automatically removed
-        // when we delete a query's target data from the reference delegate.
-        // Since this does not remove references for locally mutated documents,
-        // we have to remove the target associations for these documents
-        // manually.
-        const removed = this.localViewReferences.removeReferencesForId(
-          targetId
-        );
-        delete this.queryDataByTarget[targetId];
-        if (!keepPersistedQueryData) {
-          return PersistencePromise.forEach(removed, (key: DocumentKey) =>
-            this.persistence.referenceDelegate.removeReference(txn, key)
-          ).next(() =>
-            this.persistence.referenceDelegate.removeTarget(txn, queryData!)
-          );
-        } else {
-          return PersistencePromise.resolve();
-        }
-      });
+      const targetId = this.targetIdByQuery.get(query);
```

Comment:
We now have a map to lookup targetIds for active queries, so we no longer need to go to the QueryCache.

Reply:
Good call!